### PR TITLE
[Introspection] Detect One2One Relations on Primary Key Columns

### PR DESCRIPTION
**Pull request by do4gr**
Fixes https://github.com/prisma/prisma/issues/3237

### CODE REVIEW

File: introspection-engine/connectors/sql-introspection-connector/src/misc_helpers.rs [hunk 1]
```diff
@@ -202,15 +202,19 @@ pub(crate) fn calculate_backrelation_field(
...
                     let column_name = &relation_info.fields.first().unwrap();
-                    table.is_column_unique(column_name)
+                    table.is_column_unique(column_name) || table.is_column_primary_key(column_name)
+                }
+                _ => {
+                    table
+                        .indices
+                        .iter()
+                        .any(|i| columns_match(&i.columns, &relation_info.fields) && i.tpe == IndexType::Unique)
+                        || columns_match(&table.primary_key_columns(), &relation_info.fields)
```
Comment (tomhoule) [inline introspection-engine/connectors/sql-introspection-connector/src/misc_helpers.rs:216]: Is there a particular reason to special-case the single-column case?

**Review by tomhoule — APPROVED** [review]
looks good!

File: introspection-engine/connectors/sql-introspection-connector/src/misc_helpers.rs (hunk 1 above)
Comment (do4gr) [reply introspection-engine/connectors/sql-introspection-connector/src/misc_helpers.rs:216]: I think we do not populate the primary_key_columns if it is a singular id

Comment (do4gr) [reply introspection-engine/connectors/sql-introspection-connector/src/misc_helpers.rs:216]: Nope, checked it. You were correct it is no longer necessary to special case. The second case covers for the first one as well. I adjusted it.

**Pull request merged**